lib/workers/repository/config-migration/branch/commit-message.ts

```diff
@@ -4,43 +4,61 @@ import type { CommitMessage } from '../../model/commit-message';
 import { CommitMessageFactory } from '../../model/commit-message-factory';
 
 export class ConfigMigrationCommitMessageFactory {
-  private readonly config: RenovateConfig;
+  private commitMessage: string | null = null;
+  private prTitle: string | null = null;
 
-  private readonly configFile: string;
+  constructor(
+    private readonly config: RenovateConfig,
+    private readonly configFile: string
+  ) {}
 
-  constructor(config: RenovateConfig, configFile: string) {
-    this.config = config;
-    this.configFile = configFile;
-  }
-
-  create(): CommitMessage {
+  private create(isTitle = false): CommitMessage {
+    const messageTopic = isTitle
+      ? `Migrate renovate config`
+      : `Migrate config ${this.configFile}`;
     const { commitMessage } = this.config;
+    let { commitMessageAction, commitMessageTopic } = this.config;
 
-    this.config.commitMessageAction =
-      this.config.commitMessageAction === 'Update'
-        ? ''
-        : this.config.commitMessageAction;
+    commitMessageAction =
+      commitMessageAction === 'Update' ? '' : commitMessageAction;
 
-    this.config.commitMessageTopic =
-      this.config.commitMessageTopic === 'dependency {{depName}}'
-        ? `Migrate config ${this.configFile}`
-        : this.config.commitMessageTopic;
+    commitMessageTopic =
+      commitMessageTopic === 'dependency {{depName}}'
+        ? messageTopic
+        : commitMessageTopic;
 
-    this.config.commitMessageExtra = '';
-    this.config.semanticCommitScope = 'config';
+    const config = {
+      ...this.config,
+      semanticCommitScope: 'config',
+      commitMessageExtra: '',
+      commitMessageAction,
+      commitMessageTopic,
+    };
 
-    const commitMessageFactory = new CommitMessageFactory(this.config);
+    const commitMessageFactory = new CommitMessageFactory(config);
     const commit = commitMessageFactory.create();
 
     if (commitMessage) {
-      commit.subject = template.compile(commitMessage, {
-        ...this.config,
-        commitMessagePrefix: '',
-      });
+      config.commitMessagePrefix = '';
+      commit.subject = template.compile(commitMessage, config);
     } else {
-      commit.subject = `Migrate config ${this.configFile}`;
+      commit.subject = messageTopic;
     }
 
     return commit;
   }
+
+  getCommitMessage(): string {
+    if (this.commitMessage === null) {
+      this.commitMessage = this.create().toString();
+    }
+    return this.commitMessage;
+  }
+
+  getPrTitle(): string {
+    if (this.prTitle === null) {
+      this.prTitle = this.create(true).toString();
```

Comment:
why are we sending true/false and then
   const messageTopic = isTitle
      ? `Migrate renovate config`
      : `Migrate config ${this.configFile}`;
      
      cant we just send the title message from the start?
      then it would be more generic for future uses too

Reply:
its only used for the migration PR, therefore it might not need be that generic.

Reply:
the `isTitle` is indeed redundant and we can just send the title as an argument the the private method.